pkg/vminetworkscontroller/vmi_controller.go

```diff
@@ -56,32 +56,58 @@ func (r *VirtualMachineInstanceReconciler) Reconcile(
 	defer cancel()
 	err := r.Client.Get(contextWithTimeout, request.NamespacedName, vmi)
 	if apierrors.IsNotFound(err) {
-		r.Log.Info("could not retrieve VMI - will cleanup its IPAMClaims")
-		if err := r.Cleanup(request.NamespacedName); err != nil {
-			return controllerruntime.Result{}, fmt.Errorf("error removing the IPAMClaims finalizer: %w", err)
-		}
-		return controllerruntime.Result{}, nil
+		vmi = nil
 	} else if err != nil {
 		return controllerruntime.Result{}, err
 	}
 
-	var ownerInfo metav1.OwnerReference
 	vm := &virtv1.VirtualMachine{}
 	contextWithTimeout, cancel = context.WithTimeout(ctx, time.Second)
 	defer cancel()
-	if err := r.Client.Get(contextWithTimeout, request.NamespacedName, vm); apierrors.IsNotFound(err) {
-		r.Log.Info("Corresponding VM not found", "vm", request.NamespacedName)
-		ownerInfo = metav1.OwnerReference{APIVersion: vmi.APIVersion, Kind: vmi.Kind, Name: vmi.Name, UID: vmi.UID}
-	} else if err == nil {
-		ownerInfo = metav1.OwnerReference{APIVersion: vm.APIVersion, Kind: vm.Kind, Name: vm.Name, UID: vm.UID}
-	} else {
+	hasVMOwner := false
+	if err := r.Client.Get(contextWithTimeout, request.NamespacedName, vm); err == nil {
+		hasVMOwner = true
+	} else if !apierrors.IsNotFound(err) {
```

Comment:
I think we can just check the OwnerReferneces from the VMI and look for a VirtualMachien kind, it will more clear.

Reply:
yes and no, it is doable only if VMI != nil, otherwise we must also try to "get" VM 
so i just kept it as is for simplicity

Reply:
Ahh so we don't have always the owner list since VMI can already being delete, ok, can you add a comment realted to this ?

Reply:
hmm i can but arent we against comments?
anyhow i will add